factory/shardingFactory.go

```diff
@@ -38,7 +38,13 @@ func CreateShardCoordinator(
 			return nil, "", err
 		}
 		if selfShardId == core.DisabledShardIDAsObserver {
-			selfShardId = uint32(0)
+			pubKeyBytes, err := pubKey.ToByteArray()
+			if err != nil {
+				log.Warn("cannot get public key bytes", "error", err)
```

Comment:
I think the error escalates anyway. Might wrap the error for more context.

Reply:
done